### PR TITLE
markdown: Base.copy for Markdown.MD

### PR DESCRIPTION
**Pull request by iblislin**

### CODE REVIEW

File: test/markdown.jl [hunk 1]
```diff
@@ -1079,3 +1079,17 @@ t = """
     :-- | --:
     1   |   2"""
 @test sprint(Markdown.term, Markdown.parse(t), 0) == "a b\n– –\n1 2\n"
+
+# test Base.copy
+let
+  md = doc"test"
```
Comment (ararslan) [inline test/markdown.jl]: Should be four space indents

**Comment by KristofferC** [comment]
Out of curiosity here, when is a shallow copy like this useful? Is `deepcopy` too slow?

File: test/markdown.jl (hunk 1 above)
Comment (iblis17) [reply test/markdown.jl]: omg, I always forget it's four space indent in Base.
Can we have something like [EditorConfig](http://editorconfig.org/)?

**Comment by iblis17** [comment]
> when is a shallow copy like this useful?

I have special need for [merging docstring in MXNet](https://github.com/dmlc/MXNet.jl/pull/393/files#diff-0ca423d226049cc5744ebade856d9f1bR1092).

e.g.
```julia
doc[:foo] = doc[:foo_alias] = doc"..."
str = "moreinfo_from_c_library"
@meta_programming_on_doc foo            # and append `str` to docstring
@meta_programming_on_doc foo_alias      # if we do not copy, it got double appending.
```

> Is `deepcopy` too slow?

I tried it before, it's infeasible.

```julia
julia> s = doc"magic"
  magic

julia> deepcopy(s.meta)
ERROR: deepcopy of Modules not supported
Stacktrace:
 [1] deepcopy_internal(::Module, ::ObjectIdDict) at ./deepcopy.jl:14
 [2] deepcopy_internal(::Dict{Any,Any}, ::ObjectIdDict) at ./deepcopy.jl:91
 [3] deepcopy(::Dict{Any,Any}) at ./deepcopy.jl:8
```

**Pull request merged**